packages/core/android/src/main/java/com/datadog/reactnative/DdSdk.kt

```diff
@@ -60,8 +88,164 @@ class DdSdk(reactContext: ReactApplicationContext) : ReactContextBaseJavaModule(
+            configBuilder.useViewTrackingStrategy(ActivityViewTrackingStrategy(false))
+        } else {
+            configBuilder.useViewTrackingStrategy(NoOpViewTrackingStrategy)
+        }
+
+        val longTask =
+            (configuration.additionalConfig?.get(DD_LONG_TASK_THRESHOLD) as? Number)?.toLong()
+        if (longTask != null) {
+            configBuilder.trackLongTasks(longTask)
+        }
+
+        val firstPartyHosts =
+            (configuration.additionalConfig?.get(DD_FIRST_PARTY_HOSTS) as? List<String>)
+        if (firstPartyHosts != null) {
+            configBuilder.setFirstPartyHosts(firstPartyHosts)
+        }
+
+        buildProxyConfiguration(configuration)?.let { (proxy, authenticator) ->
+            configBuilder.setProxy(proxy, authenticator)
+        }
+
+        return configBuilder.build()
+    }
+
+    private fun buildCredentials(configuration: DdSdkConfiguration): Credentials {
+        val serviceName = configuration.additionalConfig?.get(DD_SERVICE_NAME) as? String
+        return Credentials(
+            clientToken = configuration.clientToken,
+            envName = configuration.env,
+            rumApplicationId = configuration.applicationId,
+            variant = "",
+            serviceName = serviceName
+        )
+    }
+
+    internal fun buildTrackingConsent(trackingConsent: String?): TrackingConsent {
+        return when (trackingConsent?.lowercase(Locale.US)) {
+            "pending" -> TrackingConsent.PENDING
+            "granted" -> TrackingConsent.GRANTED
+            "not_granted" -> TrackingConsent.NOT_GRANTED
+            else -> {
+                Log.w(
+                    DdSdk::class.java.canonicalName,
+                    "Unknown consent given: $trackingConsent, " +
+                        "using ${TrackingConsent.PENDING} as default"
+                )
+                TrackingConsent.PENDING
+            }
+        }
+    }
+
+    internal fun buildProxyConfiguration(configuration: DdSdkConfiguration):
+        Pair<Proxy, ProxyAuthenticator?>? {
+        val additionalConfig = configuration.additionalConfig ?: return null
+
+        val address = additionalConfig[DD_PROXY_ADDRESS] as? String
+        val port = (additionalConfig[DD_PROXY_PORT] as? Number)?.toInt()
+        val type = (additionalConfig[DD_PROXY_TYPE] as? String)?.let {
+            when (it.lowercase(Locale.US)) {
+                "http", "https" -> Proxy.Type.HTTP
+                "socks" -> Proxy.Type.SOCKS
+                else -> {
+                    Log.w(
+                        DdSdk::class.java.canonicalName,
+                        "Unknown proxy type given: $it, skipping proxy configuration."
+                    )
+                    null
+                }
+            }
+        }
+
+        val proxy = if (address != null && port != null && type != null) {
+            Proxy(type, InetSocketAddress(address, port))
+        } else {
+            return null
```

Comment:
maybe a warning log here ?

Reply:
To test it easily I would say that we wait for the tests to be added before adding this, are you ok with this?